ansible_base/lib/jwt_auth/common/auth.py

```diff
@@ -0,0 +1,203 @@
+import logging
+from urllib.parse import urljoin, urlparse
+
+import jwt
+import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+
+logger = logging.getLogger("ansible_base.lib.jwt_auth.common.auth")
+
+
+default_mapped_user_fields = [
+    "first_name",
+    "last_name",
+    "email",
+    "is_superuser",
+    "is_system_auditor",
+]
+
+
+class JWTCommonAuth:
+    def __init__(self, user_fields=default_mapped_user_fields) -> None:
+        self.mapped_user_fields = user_fields
+
+    def parse_jwt_token(self, request):
+        logger.debug("Starting JWT Authentication")
+        token = request.headers.get("X-DAB-JW-TOKEN", None)
+        if not token:
+            logger.info("X-DAB-JW-TOKEN header not set for JWT authentication")
+            return None, None
+        logger.debug(f"Received JWT auth token: {token}")
+
+        jwt_key_setting = None
+        try:
+            jwt_key_setting = settings.ANSIBLE_BASE_JWT_KEY
+        except (NameError, AttributeError):
+            logger.info("Failed to get the setting ANSIBLE_BASE_JWT_KEY")
+            return None, None
+
+        decryption_key = self.get_decryption_key(
+            jwt_key_setting,
+            validate_certs=getattr(settings, "ANSIBLE_BASE_JWT_VALIDATE_CERT", True),
+            timeout=getattr(settings, "ANSIBLE_BASE_JWT_URL_TIMEOUT", 30),
+        )
+        validated_body = self.validate_token(token, decryption_key)
+        user_model = get_user_model()
+        user, created = user_model.objects.update_or_create(
```

Comment:
I reckon this assumes the User object to be a certain "shape" (have certain fields). Wondering if the resource sync system User model comes into play here.

Reply:
We provide ways to customize what fields are in the user model but those fields would also have to be present in the JWT.